### PR TITLE
doc: Add notes to compile OVMF with legacy support

### PR DESCRIPTION
**Pull request by weltling**
This part expands on building OVMF with CSM support. 

Signed-off-by: Anatol Belski <anbelski@linux.microsoft.com>

### CODE REVIEW

File: docs/uefi.md [hunk 1]
```diff
@@ -35,8 +35,41 @@ To make Cloud Hypervisor use UEFI boot, pass the `OVMF.fd` file path as an argum
...
+echo "ACTIVE_PLATFORM=OvmfPkg/OvmfCh.dsc" >> Conf/target.txt
+echo "TARGET_ARCH=X64" >> Conf/target.txt
+echo "TOOL_CHAIN_TAG=GCC5" >> Conf/target.txt
+
+make -C ./BaseTools
+build
+
+```
+
+Please note, that the CSM support has currently only been tested with Linux guests. No legacy support is planned for Windows or other guests or whatsoever.
```
Comment (sameo) [inline docs/uefi.md:68]:
```suggestion
Please note, that the CSM support has currently only been tested with Linux guests. There are no plans to provide legacy support for other OSes (e.g. Windows).
```

File: docs/uefi.md [hunk 2]
```diff
@@ -35,8 +35,41 @@ To make Cloud Hypervisor use UEFI boot, pass the `OVMF.fd` file path as an argum
 
 The same firmware can be used with Cloud Hypervisor or with QEMU. This is particularly useful if using QEMU for the preparation phase.
 
+## Building UEFI Firmware with Compatibility Support Module (CSM)
+
+CSM is a module that allows to boot legacy operating systems using the OVMF firmware. OVMF can embed a CSM build of SeaBIOS. To build the SeaBIOS with CSM support, add `CONFIG_CSM=y` to `.config` before the build. The outcome `out/Csm16.bin` is to be moved into `OvmfPkg/Csm/Csm16/Csm16.bin` before OVMF is built. Also, the OVMF will need `-D CSM_ENABLE` to be appended, which will produce a legacy aware UEFI firmware. At the current stage, all the necessary patches are included in the Cloud Hypervisor specific [SeaBIOS branch](https://github.com/cloud-hypervisor/seabios/tree/ch). Taking into account the previous instructions, the modified command sequence to compile UEFI with the CSM support is below:
```
Comment (sameo) [inline docs/uefi.md:40]: s/`Also, the OVMF will need `-D CSM_ENABLE` to be appended,`/`Then, the OVMF build will have to be passed the  `-D CSM_ENABLE` option in order to generate a legacy aware UEFI firmware,`

Comment (sameo) [inline docs/uefi.md:40]: s/`the modified command sequence to compile UEFI with the CSM support is below:`/`the modified command sequence to compile an OVMF binary with CSM support is the following one:`

Comment (weltling) [reply docs/uefi.md]: Applied, thanks.

Comment (weltling) [reply docs/uefi.md]: Applied, thanks.

File: docs/uefi.md (hunk 1 above)
Comment (weltling) [reply docs/uefi.md]: Applied, thanks.

**Review by sameo — APPROVED** [review]

**Pull request merged**